keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/plugin/ResolvableResourceHandler.kt

```diff
@@ -60,10 +62,17 @@ interface ResolvableResourceHandler<S : Any, R : Any> : KeelPlugin {
         e
       )
     }
+    val app = when (spec) {
+      is Monikered -> spec.moniker.app
+      is Applicationed -> spec.application
+      else -> "noapp" // todo eb: is this the right way to handle this?
```

Comment:
feels like if app is really optional then maybe modelling it as such on metadata would be better. Right now I'm domain squatting on the `noapp` application in our spinnaker just because I'm a jerk

Reply:
I'd expect nothing less of you.

I'm changing this to throw an exception here to indicate there's something wrong with the plugin if it allows a resource which doesn't have an application.